src/net_processing.cpp

```diff
@@ -2228,6 +2228,7 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
         // much work as our tip, download as much as possible.
         if (fCanDirectFetch && pindexLast->IsValid(BLOCK_VALID_TREE) && chainActive.Tip()->nChainWork <= pindexLast->nChainWork) {
             vector<const CBlockIndex *> vToFetch;
+            assert(pindexLast);
```

Comment:
Yes, you're absolutely right - the redundant check `pindexWalk`-check fooled me as mentioned in [this comment](https://github.com/bitcoin/bitcoin/pull/9559#issuecomment-272669553) :-)  Sorry for the noise. Closing this PR!